tuf/repository_tool.py

```diff
@@ -1814,12 +1880,18 @@ def add_delegated_role(self, rolename, targets_object):
       raise securesystemslib.exceptions.FormatError(repr(targets_object) + ' is'
           ' not a Targets object.')
 
-
-    if rolename in self._delegated_roles:
-      logger.debug(repr(rolename) + ' already exists.')
+    if succinct:
+      if rolename in self._succinct_delegations:
+        logger.debug(repr(rolename) + ' already exists.')
```

Comment:
Do we really want to ignore this? What happens if we load a repository with succinct hashed bin delegations and want to change i.e. the number of succinct hashed bins?

Is there any harm in adding the role to `self._succinct_delegations` even when it's already present?

Reply:
You're right, I think replacing any existing value makes more sense.